src/dev/amdgpu/amdgpu_device.cc

```diff
@@ -521,6 +521,7 @@ AMDGPUDevice::writeDoorbell(PacketPtr pkt, Addr offset)
         RequestPtr pending_req(pkt->req);
         PacketPtr pending_pkt = Packet::createWrite(pending_req);
         uint8_t *pending_data = new uint8_t[pkt->getSize()];
+        memcpy(pending_data, pkt->getPtr<uint8_t>(), pkt->getSize());
```

Comment:
Can you please add a comment explaining this?  I'm assuming it's copying doorbell information into the packet?  But why is it needed?

Reply:
As it is now, the packet data is uninitialized and the random value read when replaying the packet won't write to the correct doorbell.  This causes the queue to not be woken up, followed by the application deadlocking